.github/workflows/tests.yml

```
@@ -28,4 +28,20 @@ jobs:
       run: go vet -v ./...
 
     - name: Test
-      run: go test -v ./...
\ No newline at end of file
+      run: go test -v ./...
+
+  vet-adapters:
+    strategy:
+      matrix:
+        adapters: [ 'kafkastreamer', 'reflexstreamer', 'sqlstore', 'sqltimeout', 'jlog' ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.22'
+
+      - name: Vet
+        run: cd ./adapters/${{ matrix.adapters }} && go vet -v ./...
```

Would something more dynamic work, so that we don't have to remember to add/remove adapters from the matrix? 
Maybe:
```suggestion
        run: go list -f './adapters' -m | xargs go vet
```

---

That could work. The reason that they are listed is that these adapters define their own go module whereas all the others (in-mem adapters) form part of the `github.com/luno/workflow` module

---

When I ran the one from my other suggestion, the output included:
```
github.com/luno/workflow/internal/metrics
github.com/luno/workflow
github.com/luno/workflow/adapters/memrecordstore
github.com/luno/workflow/adapters/memtimeoutstore
github.com/luno/workflow/adapters/memstreamer
github.com/luno/workflow/adapters/adaptertest
github.com/luno/workflow/adapters/memrecordstore_test
github.com/luno/workflow/adapters/memrolescheduler_test
github.com/luno/workflow/adapters/memtimeoutstore_test
github.com/luno/workflow/adapters/memstreamer_test
```
So think it's OK!